sys-libs/libcxx/libcxx-3.9.0.ebuild

```diff
@@ -0,0 +1,161 @@
+	https://github.com/llvm-mirror/libcxx.git"
+
+[ "${PV%9999}" != "${PV}" ] && SCM="git-r3" || SCM=""
+
+inherit ${SCM} cmake-multilib toolchain-funcs
+
+DESCRIPTION="New implementation of the C++ standard library, targeting C++11"
+HOMEPAGE="http://libcxx.llvm.org/"
+if [ "${PV%9999}" = "${PV}" ] ; then
+	SRC_URI="http://llvm.org/releases/${PV}/${P}.src.tar.xz"
+	S="${WORKDIR}/${P}.src"
+else
+	SRC_URI=""
+fi
+
+LICENSE="|| ( UoI-NCSA MIT )"
+SLOT="0"
+if [ "${PV%9999}" = "${PV}" ] ; then
+	KEYWORDS="~amd64 ~mips ~x86 ~amd64-fbsd ~x86-fbsd ~amd64-linux ~x86-linux"
+else
+	KEYWORDS=""
+fi
+IUSE="elibc_glibc elibc_musl +libcxxrt libunwind +static-libs test"
+REQUIRED_USE="libunwind? ( libcxxrt )"
+
+RDEPEND="libcxxrt? ( sys-libs/libcxxrt[libunwind=,static-libs?,${MULTILIB_USEDEP}] )
+	!libcxxrt? ( >=sys-devel/gcc-4.7:=[cxx] )"
+DEPEND="${RDEPEND}
+	test? ( sys-devel/clang )
+	app-arch/xz-utils"
+
+DOCS=( CREDITS.TXT )
+
+PATCHES=(
+	# Back-port of https://reviews.llvm.org/D23232, allowing building both
+	# shared and static libs in one run.
+	"${FILESDIR}/${PN}-3.9-cmake-static-lib.patch"
+
+	# Add link flag "-Wl,-z,defs" to avoid underlinking; this is needed in a
+	# out-of-tree build.
+	"${FILESDIR}/${PN}-3.9-cmake-link-flags.patch"
+)
+
+pkg_setup() {
+	if ! use libcxxrt && ! tc-is-gcc ; then
+		eerror "To build ${PN} against libsupc++, you have to use gcc. Other"
+		eerror "compilers are not supported. Please set CC=gcc and CXX=g++"
+		eerror "and try again."
+		die
+	fi
+	if tc-is-gcc && [[ $(gcc-version) < 4.7 ]] ; then
+		eerror "${PN} needs to be built with gcc-4.7 or later (or other"
+		eerror "conformant compilers). Please use gcc-config to switch to"
+		eerror "gcc-4.7 or later version."
+		die
+	fi
+}
+
+multilib_src_configure() {
+	local cxxabi cxxabi_incs
+	if use libcxxrt; then
+		cxxabi=libcxxrt
+		cxxabi_incs="${EPREFIX}/usr/include/libcxxrt"
+	else
+		local gcc_inc="${EPREFIX}/usr/lib/gcc/${CHOST}/$(gcc-fullversion)/include/g++-v$(gcc-major-version)"
+		cxxabi=libsupc++
+		cxxabi_incs="${gcc_inc};${gcc_inc}/${CHOST}"
+	fi
+
+	local libdir=$(get_libdir)
+	local mycmakeargs=(
+		-DLIBCXX_LIBDIR_SUFFIX=${libdir#lib}
+		-DLIBCXX_ENABLE_SHARED=ON
+		-DLIBCXX_ENABLE_STATIC=$(usex static-libs)
+		-DLIBCXX_CXX_ABI=${cxxabi}
+		-DLIBCXX_CXX_ABI_INCLUDE_PATHS=${cxxabi_incs}
+		-DLIBCXX_ENABLE_ABI_LINKER_SCRIPT=OFF
+		-DLIBCXX_HAS_MUSL_LIBC=$(usex elibc_musl)
+		-DLIBCXX_HAS_GCC_S_LIB=$(usex !libunwind)
+		-DCMAKE_SHARED_LINKER_FLAGS=$(usex libunwind "-lunwind" "")
+	)
+	cmake-utils_src_configure
+}
+
+# Tests fail for now, if anybody is able to fix them, help is very welcome.
+multilib_src_test() {
+	cd "${BUILD_DIR}/test"
+	LD_LIBRARY_PATH="${BUILD_DIR}/lib:${LD_LIBRARY_PATH}" \
+		CC="clang++ $(get_abi_CFLAGS) ${CXXFLAGS}" \
+		HEADER_INCLUDE="-I${BUILD_DIR}/include" \
+		SOURCE_LIB="-L${BUILD_DIR}/lib" \
+		LIBS="-lm $(usex libcxxrt -lcxxrt "")" \
+		./testit || die
+	# TODO: fix link against libsupc++
```

Comment:
Also note that you might need to checkout LLVM sources for tests to work, and then use `-DLLVM_MAIN_SRC_DIR` or alike to override the source directory.